addons/event/models/res_partner.py

```diff
@@ -13,7 +13,7 @@ def _compute_event_count(self):
         if not self.user_has_groups('event.group_event_user'):
             return
         for partner in self:
-            partner.event_count = self.env['event.event'].search_count([('registration_ids.partner_id', 'child_of', partner.ids)])
+            partner.event_count = self.env['event.event'].sudo().search_count([('registration_ids.partner_id', 'child_of', partner.ids)])
```

Comment:
Why do you need a sudo here ?

If you don't have access to events, you should simply hide the button that trigger the Access Error nope?